server/datastore/mysql/software.go

```diff
@@ -213,9 +213,9 @@ func (d *Datastore) hostSoftwareFromHostID(tx *sqlx.Tx, id uint) ([]fleet.Softwa
 	sql := `
 		SELECT s.id, s.name, s.version, s.source, coalesce(scp.cpe, "") as generated_cpe, 
 			IF(
-				JSON_ARRAYAGG(scv.cve) = JSON_ARRAYAGG(null), 
+				GROUP_CONCAT(scv.cve) = null, 
```

Comment:
You could use COUNT() here as it seems it is used only to check if all occurrences of `scv.cve` are NULL.
 
```suggestion
				COUNT(scv.cve) = 0, 
```

Reply:
Oh, great idea!